tests/ha/test_control_pod_restart.py

```diff
@@ -0,0 +1,417 @@
+        cls.deploy_lc_obj = ProvDeployK8sCortxLib()
+        cls.s3_clean = cls.test_prefix = cls.random_time = None
+        cls.s3acc_name = cls.s3acc_email = cls.bucket_name = cls.object_name = None
+        cls.restore_pod = cls.deployment_backup = cls.deployment_name = cls.restore_method = None
+        cls.restore_node = cls.node_name = cls.deploy = None
+        cls.restore_ip = cls.node_iface = cls.new_worker_obj = cls.node_ip = None
+        cls.mgnt_ops = ManagementOPs()
+        cls.system_random = secrets.SystemRandom()
+        cls.motr_obj = MotrCoreK8s()
+        cls.rest_iam_user = RestIamUser()
+
+        for node in range(cls.num_nodes):
+            cls.host = CMN_CFG["nodes"][node]["hostname"]
+            cls.username.append(CMN_CFG["nodes"][node]["username"])
+            cls.password.append(CMN_CFG["nodes"][node]["password"])
+            if CMN_CFG["nodes"][node]["node_type"] == "master":
+                cls.host_master_list.append(cls.host)
+                cls.node_master_list.append(LogicalNode(hostname=cls.host,
+                                                        username=cls.username[node],
+                                                        password=cls.password[node]))
+                cls.hlth_master_list.append(Health(hostname=cls.host,
+                                                   username=cls.username[node],
+                                                   password=cls.password[node]))
+            else:
+                cls.host_worker_list.append(cls.host)
+                cls.node_worker_list.append(LogicalNode(hostname=cls.host,
+                                                        username=cls.username[node],
+                                                        password=cls.password[node]))
+
+        cls.rest_obj = S3AccountOperations()
+        cls.s3_mp_test_obj = S3MultipartTestLib(endpoint_url=S3_CFG["s3_url"])
+        cls.test_file = "ha-mp_obj"
+        cls.test_dir_path = os.path.join(TEST_DATA_FOLDER, "HATestMultipartUpload")
+        control_pods = cls.node_master_list[0].get_pods_node_fqdn(const.CONTROL_POD_NAME_PREFIX)
+        ctrl_pod = list(control_pods.keys())[0]
+        backup_path = cls.node_master_list[0].backup_deployment(
+            deployment_name=cls.node_master_list[0].get_deploy_replicaset(ctrl_pod)[1])[1]
+        cls.original_backup = backup_path.split(".")[0] + "_original.yaml"
+        cls.node_master_list[0].rename_file(old_filename=backup_path,
+                                            new_filename=cls.original_backup)
+        cls.original_control_node = control_pods.get(ctrl_pod)
+        cls.multipart_obj_path = None
+
+    def setup_method(self):
+        """
+        This function will be invoked prior to each test case.
+        """
+        LOGGER.info("STARTED: Setup Operations")
+        self.random_time = int(time.time())
+        self.restore_node = False
+        self.restore_ip = False
+        self.deploy = False
```

Comment:
do we need to redeploy?

Reply:
In case of restart node, we will have to redeploy.